frontend/src/types/schemaSystem.ts

```diff
@@ -0,0 +1,282 @@
+import { SchemaReviewId } from "./id";
+import { Principal } from "./principal";
+
+export enum RuleLevel {
+  Disabled = "disabled",
+  Error = "error",
+  Warning = "warning",
+}
+
+export const LEVEL_LIST = [
+  RuleLevel.Error,
+  RuleLevel.Warning,
+  RuleLevel.Disabled,
+];
+
+enum PayloadType {
+  String = "string",
+  StringArray = "string[]",
+  Template = "template",
+}
+
+interface StringPayload {
+  type: PayloadType.String;
+  default: string;
+  value?: string;
+}
+
+interface StringArrayPayload {
+  type: PayloadType.StringArray;
+  default: string[];
+  value?: string[];
+}
+
+interface TemplatePayload {
+  type: PayloadType.Template;
+  default: string;
+  templates: { id: string; description?: string }[];
+  value?: string;
+}
+
+export interface RulePayload {
+  [key: string]: StringPayload | StringArrayPayload | TemplatePayload;
+}
+
+export type DatabaseType = "MySQL" | "Common";
+
+export type CategoryType = "engine" | "naming" | "query" | "table" | "column";
+
+export interface SchemaRule {
+  id: string;
+  category: CategoryType;
+  database: DatabaseType[];
```

Comment:
Let's make this a singular. In the short term, we only have mysql, postgres, common, so any rule would fit in either of those 3 categories. And if in the future, we have a rule which applies to mysql, clickhouse, but not for the others, then we can duplicate the rule.